samples/jobs/describe_job_execution/main.cpp

```diff
@@ -212,92 +206,83 @@ int main(int argc, char *argv[])
         exit(-1);
     }
 
-    std::mutex mutex;
-    std::unique_lock<std::mutex> uniqueLock(mutex);
-    conditionVariable.wait(uniqueLock, [&]() { return connectionSucceeded || connectionClosed; });
-
-    if (connectionSucceeded)
+    if (connectionCompletedPromise.get_future().get())
     {
         IotJobsClient client(connection);
 
         DescribeJobExecutionSubscriptionRequest describeJobExecutionSubscriptionRequest;
         describeJobExecutionSubscriptionRequest.ThingName = thingName;
         describeJobExecutionSubscriptionRequest.JobId = jobId;
 
-        // This isn't absolutely neccessary but since we're doing a publish almost immediately afterwards,
+        // This isn't absolutely necessary but since we're doing a publish almost immediately afterwards,
         // to be cautious make sure the subscribe has finished before doing the publish.
-        auto subAckHandler = [&](int ioErr) {
-            if (!ioErr)
-            {
-                conditionVariable.notify_one();
-            }
+        std::promise<void> subAckedPromise;
+        auto subAckHandler = [&](int) {
+            /* if error code returns it will be recorded by the other callback */
+            subAckedPromise.set_value();
         };
-
         auto subscriptionHandler = [&](DescribeJobExecutionResponse *response, int ioErr) {
             if (ioErr)
             {
                 fprintf(stderr, "Error %d occurred\n", ioErr);
-                conditionVariable.notify_one();
```

Comment:
[SubscribeToDescribeJobExecutionAccepted](https://github.com/aws/aws-iot-device-sdk-cpp-v2/blob/master/jobs/source/IotJobsClient.cpp#L150) subAckHandler will always be called. I think we can probably safely wait for that callback instead of either one.